exporter/metric/prometheus/prometheus.go

```diff
@@ -0,0 +1,262 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheus
+
+import (
+	"context"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"go.opentelemetry.io/otel/sdk/export/metric/aggregator"
+	"net/http"
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"go.opentelemetry.io/otel/api/core"
+	export "go.opentelemetry.io/otel/sdk/export/metric"
+)
+
+const (
+	prefixSplitter = "+"
+)
+
+type metricID *export.Descriptor
+
+// Exporter is an implementation of metric.Exporter that sends metrics to
+// Prometheus.
+type Exporter struct {
+	sync.RWMutex
+	registerer prometheus.Registerer
+	gatherer   prometheus.Gatherer
+
+	counters map[metricID]prometheus.Counter
+	gauges   map[metricID]prometheus.Gauge
+
+	onRegisterError func(e error)
+	counterVecs     map[string]*prometheus.CounterVec
```

Comment:
I think these maps **should** use the `*export.Descriptor`, unless I'm really misunderstanding. The `CounterVec` type corresponds to an instrument (use `*export.Descriptor`), the `Counter` type corresponds to a handle (descriptor+encoded labels).

Reply:
Ok, yes. That's right. I got them inverted. 

That's going to be a bad performance bottleneck though. If every time we have to look up a `Counter` by encoding labels that's just too heavy.. We probably do not need the descriptor if we use the labels.. No?

Reply:
I don't think so @shreyassrivatsan, two different counters/gauge/other metrics could have the same exact labels, so we do need the descriptor as well.

`Labels` has an `encoded string` so it doesn't need to encode it every time. Please correct if I am wrong @jmacd. 

btw, I have already changed the `CounterVec` to use `*export.Descriptor` and `Counter` to use the descriptor+encoded labels.